src/mattermostdriver/websocket.py

```diff
@@ -77,13 +79,14 @@ async def _start_loop(self, websocket, event_handler):
         keep_alive = asyncio.ensure_future(self._do_heartbeats(websocket))
         log.debug("Waiting for messages on websocket")
         while self._alive:
-            message = await websocket.recv()
+            message = await websocket.receive_str()
             self._last_msg = time.time()
             await event_handler(message)
         log.debug("cancelling heartbeat task")
         keep_alive.cancel()
         try:
             await keep_alive
+            await websocket.close()
```

Comment:
Seemed with websockets the connection was never cleanly closed. We now do so.

Reply:
Don’t do this here.  The above await always raises CancelledError

Reply:
Also I would use the async context manager so you don’t have to explicitly  close the websocket object

Reply:
I do use the context manager so probably don't need this at all ...

Reply:
Kept it, as it would be nice to cleanly close the websocket then the aiohttp session that the context manager takes care of.